### PR TITLE
Don't print empty observables

### PR DESCRIPTION
**Pull request by wkotlarski**
This PR ensures that observables are not printed to SLHA output if they are not computed. Let me know what you think.

### CODE REVIEW

File: templates/librarylink.cpp.in [hunk 1]
```diff
@@ -660,15 +660,18 @@ void @ModelName@_spectrum_impl<Solver_type>::calculate_model_observables(
...
+       const Spectrum_generator_settings& settings) const
 {
    const auto& problems = std::get<0>(models).get_problems();
    const auto force_output = std::get<0>(models).do_force_output();
 
    slha_io.set_spinfo(problems);
    if (!problems.have_problem() || force_output) {
       slha_io.set_spectrum(models);
-      slha_io.set_extra(std::get<0>(models), scales, observables);
+      if (settings.get(Spectrum_generator_settings::calculate_observables)) {
```
Comment (Expander) [inline templates/librarylink.cpp.in:672]: I think here you're not only disabling the observables, but also the output of the 'scales'.

File: templates/run.cpp.in [hunk 2]
```diff
@@ -103,7 +103,9 @@ int run_solver(flexiblesusy::@ModelName@_slha_io& slha_io,
             spectrum_generator_settings.get(
                Spectrum_generator_settings::force_positive_masses));
          slha_io.set_spectrum(models);
-         slha_io.set_extra(std::get<0>(models), scales, observables);
+         if (spectrum_generator_settings.get(Spectrum_generator_settings::calculate_observables)) {
```
Comment (Expander) [inline templates/run.cpp.in:106]: see my comment above

**Comment by Expander** [comment]
I like the idea, as it might be clearer that printing 0. Please see my comments in the code.

**Comment by wkotlarski** [comment]
I wanted to only disable observables but this is not possible. The reason is that what we're actually printing is the `ExtraSLHAOutputBlocks` block. We'd have to split it or, slternatively, pass settings to `set_extra`.

**Comment by Expander** [comment]
Yes, you are right. However, since there may be anything in `ExtraSLHAOutputBlocks`, I find it problematic to disable everything when the observables are disabled.

I find your proposal of passing the `settings` to `set_extra` better.

**Comment by pathron** [comment]
I agree with Alex we added the scales output because it was wanted in some cases and this does not depend on whether we calculate the observables or not. Passing settings to set_extra seems reasonable to me.

**Comment by wkotlarski** [comment]
How about now?

**Comment by Expander** [comment]
I like this! But where is the disabling of the observables now?

**Comment by wkotlarski** [comment]
I wasn't finished yet. Now it should be fine.

**Comment by Expander** [comment]
That looks OK to me.

**Comment by wkotlarski** [comment]
Thanks!

**Pull request merged**